x-pack/plugins/security/public/management/users/edit_user/edit_user_page.tsx

```diff
@@ -84,7 +87,17 @@ class EditUserPageUI extends Component<Props, State> {
   }
 
   public async componentDidMount() {
-    const { username, apiClient, securitySetup } = this.props;
+    await this.setCurrentUser();
+  }
+
+  public async componentDidUpdate(prevProps: Props) {
```

Comment:
**question** Is this logic needed as a result of the new routing we have to do?

Reply:
Yeah, when you open Edit User page and manually change username in the URL or just remove it and hit `Enter`, Router logic will be triggered and for example will update breadcrumbs. But since view is already mounted we need to reload user as well.

That made me realize that we need `componentDidUpdate` for Edit Role Mappings as well, I'll add it. Edit Role page is based on Hooks so it already should work properly.